transit_odp/publish/forms/seasonal_services.py

```diff
@@ -180,13 +202,48 @@ def clean_registration_code(self):
             )
         return registration_code
 
-    def clean(self):
-        cleaned_data = self.cleaned_data
-        start = cleaned_data.get("start")
-        end = cleaned_data.get("end")
-        if start is None or end is None:
-            return cleaned_data
 
-        if start > end:
-            raise ValidationError("Start date must be earlier than end date")
-        return cleaned_data
+class EditDateForm(BaseForm):
+    def __init__(self, *args, **kwargs):
+        self.org_id = kwargs.pop("org_id", None)
+        self.page = kwargs.pop("page", None)
+        super().__init__(*args, **kwargs)
+
+    def get_layout(self):
+        help_modal = render_to_string(
+            "publish/snippets/help_modals/seasonal_services.html"
+        )
+        edit_snippet = render_to_string(
+            "publish/seasonal_services/edit_page_service_code_snippet.html",
+            context={"form": self},
+        )
+
+        return Layout(
+            HTML(edit_snippet),
+            HTML(
+                format_html(
+                    '<h2 class="govuk-heading-s">Service operating dates {}</h2>',
+                    help_modal,
+                ),
+            ),
+            DateDiv(
+                FieldNoErrors("start", wrapper_class="date"),
+                FieldNoErrors("end", wrapper_class="date"),
+                css_class="date-container",
+            ),
+            SEPARATOR,
+            ButtonHolder(
+                SAVE_BUTTON,
+                cancel_seasonal_service(self.org_id, self.page),
+                css_class="buttons",
+            ),
+        )
+
+
+class DeleteForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        self.seasonal_service_id = kwargs.pop("seasonal_service_id", None)
+        super().__init__(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        return SeasonalService.objects.get(id=self.seasonal_service_id).delete()
```

Comment:
then seasonal_service_id will be in `self.cleaned_data`

Reply:
there is  a `cleaned_data` but it's empty, and has to be empty, because there is no form, there is no template too!!
For the un-validated form  there is another dictionary `data`